base_layer/core/src/chain_storage/blockchain_database.rs

```diff
@@ -247,16 +250,50 @@ where B: BlockchainBackend
             txn = DbTransaction::new();
             blockchain_db.insert_block(genesis_block.clone())?;
             let body = &genesis_block.block().body;
-            let utxo_sum = body.outputs().iter().map(|k| &k.commitment).sum::<Commitment>();
+            let input_sum = body
+                .inputs()
+                .iter()
+                .map(|k| k.commitment())
+                .collect::<Result<Vec<_>, _>>()?
+                .into_iter()
+                .sum::<Commitment>();
+            let output_sum = body.outputs().iter().map(|k| &k.commitment).sum::<Commitment>();
+            let total_utxo_sum = &output_sum - &input_sum;
             let kernel_sum = body.kernels().iter().map(|k| &k.excess).sum::<Commitment>();
             txn.update_block_accumulated_data(*genesis_block.hash(), UpdateBlockAccumulatedData {
                 kernel_sum: Some(kernel_sum.clone()),
                 ..Default::default()
             });
             txn.set_pruned_height(0);
-            txn.set_horizon_data(kernel_sum, utxo_sum);
+            txn.set_horizon_data(kernel_sum, total_utxo_sum);
             blockchain_db.write(txn)?;
             blockchain_db.store_pruning_horizon(config.pruning_horizon)?;
+            // Sanity check
+            match blockchain_db.fetch_block_by_hash(*genesis_block.hash(), false)? {
+                Some(block) => {
+                    let header_from_db = block.block().clone().header;
+                    let header_from_code = genesis_block.block().clone().header;
+                    if header_from_db != header_from_code {
+                        return Err(ChainStorageError::CriticalError(
+                            "Genesis block header  in db does not match genesis block in code".into(),
+                        ));
+                    }
+                    let mut body_from_db = block.block().clone().body;
+                    body_from_db.sort();
+                    let mut body_from_code = genesis_block.block().clone().body;
+                    body_from_code.sort();
+                    if body_from_db != body_from_code {
+                        return Err(ChainStorageError::CriticalError(
+                            "Genesis block body in db does not match genesis block in code".into(),
+                        ));
+                    }
+                },
+                None => {
+                    return Err(ChainStorageError::CriticalError(
+                        "Genesis block could not be created".into(),
+                    ))
+                },
+            }
```

Comment:
Why add this, you are checking if what you just added is in the database?
If that would have failed, it would have thrown an error in ln 269
All this is doing is adding more lines. This code should assume that the lmdb code is working. We should assume that lmdb has its own unit tests to ensure if a write fails, it will error. Or that what you write is actually in the database